lib/workos/deprecated_hash_wrapper.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module WorkOS
+  # A Temporary wrapper class for a Hash, currently the base class for
+  # WorOS::DirectoryGroup and WorkOS::DirectoryUser. Makes all the Hash
+  # methods available to those classes, but will also emit a deprecation
+  # warning whenever any of them are used.
+  #
+  # Once we deprecate Hash compatibility, this model can be deleted.
+  class DeprecatedHashWrapper < Hash
+    (public_instance_methods - Object.methods).each do |method_name|
+      define_method method_name do |*args, &block|
+        print_deprecation_warning(method_name)
+        super(*args, &block)
+      end
+    end
+
+    def [](attribute_name)
+      print_deprecation_warning('[]', attribute_name)
+
+      super(attribute_name.to_sym)
+    end
+
+    private
+
+    def print_deprecation_warning(method_name, param_name = nil)
+      class_name = self.class.name
+      usage = if method_name == '[]'
+                "#{class_name.downcase}.#{param_name}"
+              else
+                "#{class_name.downcase}.#{method_name}"
+              end
+
+      warning_message = "WARNING: The Hash style access for #{class_name} attributes is deprecated
+and will be removed in a future version. Please use `#{usage}` or equivalent accessor.\n"
```

Comment:
The resulting warning looks good for `[]`, but for other `Hash` methods, the advice we give doesn't make sense. For example, here's the warning for calling `Hash#replace`:

```
WARNING: The Hash style access for WorkOS::DirectoryUser attributes is deprecated
and will be removed in a future version. Please use `workos::directoryuser.replace` or equivalent accessor.
```

Reply:
🤔, yeah, that's fair, I was trying to come up with something that would sound reasonable for any method, but that's clearly not doing it... Let me see if I can come up with something better!

Reply:
Ok, I went ahead and broke up the warnings to just have separate ones for `[]` and all the other methods:

`WARNING: The Hash style access for WorkOS::DirectoryUser attributes is deprecated and will be removed in a future version. Please use 'user.id' or equivalent accessor.`

`WARNING: Hash compatibility for WorkOS::DirectoryUser is deprecated and will be removed in a future version. Please use 'user.to_hash.default' to access methods on the attribute Hash object.`

Getting a single warning message for all cases just wasn't ending up with anything that I liked. There might be cases where that second message is also not quite perfect, but I'm hoping/assuming that most of the uses are really just for `[]`, so if there are some methods where that message doesn't quite work, they're not getting called very much.

@mthadley, does that all look reasonable? I also went ahead and added a way to call the `replace` the method from the initializer that won't trigger the warning: https://github.com/workos/workos-ruby/pull/148/files#diff-c5fd131b255f20d0db179e1dcefa1ddbc01f523395d64dfd9141d4ce804921adR20-R22